src/main/java/net/mcreator/ui/minecraft/ProcedureSelector.java

```diff
@@ -117,6 +117,9 @@ public ProcedureSelector(@Nullable IHelpContext helpContext, MCreator mcreator,
 			defaultName = "(always)";
 			setBorder(BorderFactory
 					.createLineBorder(new Dependency("", VariableElementType.LOGIC.toDependencyType()).getColor()));
+		} else if(returnType == VariableElementType.STRING){
+			setBorder(BorderFactory
+					.createLineBorder(new Dependency("", VariableElementType.STRING.toDependencyType()).getColor()));

```

Comment:
There is now a static method in Dependency to get color so no need to initialize an object. You can correct logic while you are at it. Not really your fault, but noticed this could be improved ;)

Reply:
Also missing space between ) and {